### ISSUE TITLE
Update tests to be more resilient against assert message changes in Node core

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#10111]
Our tests for `RuleTester` currently make assertions about the error messages that `RuleTester` outputs. These tests have been broken a few times in the past due to changes in the error messages created by Node's `assert` module (e.g. https://github.com/eslint/eslint/pull/9688, https://github.com/nodejs/node/pull/19467#issuecomment-374669129). Having the tests break is generally inconvenient (among other things, Node runs the `eslint` tests as part of [citgm](https://github.com/nodejs/citgm) in order to detect regressions), so we should try to make these tests more robust.

One way to do it would be to dynamically generate an expected error message as part of the test. For example, we could replace the code [here](https://github.com/eslint/eslint/blob/0360cc25c86619d30e37e25d4ce9a78309591c18/tests/lib/testers/rule-tester.js#L117-L130):

```js
assert.throws(() => {
    ruleTester.run(/* ... */)
}, /Bad var\.((.*==)|(.*strictEqual)).*Bad error message/);
```

...with something like this:

```js
const nodeAssert = require("assert");

assert.throws(() => {
    ruleTester.run(/* ... */)
}, new nodeAssert.AssertionError({ actual: "Bad var.", expected: "Bad error message", operator: "===" }).message);
```
